microservices-connector/cmd/router/main.go

```diff
@@ -477,6 +479,112 @@ func mcGraphHandler(w http.ResponseWriter, req *http.Request) {
+			if err != nil {
+				http.Error(w, "Failed to parse multipart form", http.StatusBadRequest)
+				return
+			}
+			// Create a buffer to hold the new form data
+			var buf bytes.Buffer
+			writer := multipart.NewWriter(&buf)
+			// Copy all form fields from the original request to the new request
+			for key, values := range r.MultipartForm.Value {
+				for _, value := range values {
+					err := writer.WriteField(key, value)
+					if err != nil {
+						http.Error(w, "Failed to write form field", http.StatusInternalServerError)
+						return
+					}
+				}
+			}
+			// Copy all files from the original request to the new request
+			for key, fileHeaders := range r.MultipartForm.File {
+				for _, fileHeader := range fileHeaders {
+					file, err := fileHeader.Open()
+					if err != nil {
+						http.Error(w, "Failed to open file", http.StatusInternalServerError)
+						return
+					}
+					defer func() {
+						if err := file.Close(); err != nil {
+							log.Error(err, "error closing file")
+						}
+					}()
+					part, err := writer.CreateFormFile(key, fileHeader.Filename)
+					if err != nil {
+						http.Error(w, "Failed to create form file", http.StatusInternalServerError)
+						return
+					}
+					_, err = io.Copy(part, file)
+					if err != nil {
+						http.Error(w, "Failed to copy file", http.StatusInternalServerError)
+						return
+					}
+				}
+			}
+
+			defer func() {
+				err = writer.Close()
+				if err != nil {
+					http.Error(w, "Failed to close writer", http.StatusInternalServerError)
+					return
+				}
+			}()
+			req, err := http.NewRequest(r.Method, serviceURL, &buf)
+			if err != nil {
+				http.Error(w, "Failed to create new request", http.StatusInternalServerError)
+				return
+			}
+			// Copy headers from the original request to the new request
+			for key, values := range r.Header {
+				for _, value := range values {
+					req.Header.Add(key, value)
+				}
+			}
+			req.Header.Set("Content-Type", writer.FormDataContentType())
+			client := &http.Client{}
+			resp, err := client.Do(req)
+			if err != nil {
+				http.Error(w, "Failed to send request to backend", http.StatusInternalServerError)
+				return
+			}
+			defer func() {
+				if err := resp.Body.Close(); err != nil {
+					log.Error(err, "error closing response body stream")
+				}
+			}()
+			// Copy the response headers from the backend service to the original client
+			for key, values := range resp.Header {
+				for _, value := range values {
+					w.Header().Add(key, value)
+				}
+			}
+			w.WriteHeader(resp.StatusCode)
+			// Copy the response body from the backend service to the original client
+			_, err = io.Copy(w, resp.Body)
+			if err != nil {
+				log.Error(err, "failed to copy response body")
+			}
+		}
+	}
+}
+
+func initializeRoutes() *http.ServeMux {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", mcGraphHandler)
+	mux.HandleFunc("/dataprep", mcDataHandler)
```

Comment:
To @irisdingbj comment about expecting only one dataprep path and @zhlsunshine allowing/expecting more, is this logic a simplification of adding route only for one dataprep path. We have to be consistent in our logic, support only one now and later more perhaps, or support multiple everywhere.

Reply:
Let's target support more dataprep, in this case, we need generate handler dynamically instead of the fixed /dataprep one.

Reply:
my thought is like other http routers:
	mux.HandleFunc("/", mcGraphHandler)
	mux.HandleFunc("/dataprep", mcDataHandler)
        mux.HandleFunc("/*other invalid path*", helper) => indicate the legal paths

Reply:
according to my search, in the default go router :
    // Catch-all route for handling invalid paths
    router.NoRoute(func(c *gin.Context) {
        c.String(404, "Page not found")
    })

if that works, my thought is maybe we can add a helper here to indicate the correct path to access

Reply:
Hi @KfreeZ, for now, we can just a remind for for user. Like in PR description:
![image](https://github.com/user-attachments/assets/b0db322d-35f7-4532-bdcf-a871def9b8de)